sorc/ncep_post.fd/UPP_PHYSICS.f

```diff
@@ -1320,6 +1320,9 @@ SUBROUTINE CALCAPE2(ITYPE,DPBND,P1D,T1D,Q1D,L1D,    &
               THUNDER(I,J) = .FALSE.
             ENDIF
           ENDIF
+
+          ! Limit LCL to prevent out-of-bounds accesses later
+          LCL(I,J) = max(min(LCL(I,J),LM-1),1)
```

Comment:
@SamuelTrahanNOAA Please add a change log with your name at the beginning session of this subroutine.

Reply:
Done. I added one to the other file too.